lib/io/src/spi.rs

```diff
@@ -0,0 +1,22 @@
+/// SPI errors that can occur
+/// From: https://docs.embassy.dev/embassy-stm32/git/stm32f103c8/spi/enum.Error.html
+pub enum SpiError {
```

Comment:
is there a reason to redefine this instead of doing `pub use embassy_stm32::spi::Error as SpiError;` (i think that's the right way to reexport the type)

Reply:
Part of the bind we're having with abstractions is that we can't use `embassy_stm32` because we have to enable exactly one board feature making the crate board-specific (so we wouldn't be able to reuse it across different boards)